irohad/main/application.cpp

```diff
@@ -131,35 +131,40 @@ Irohad::~Irohad() {
 /**
  * Initializing iroha daemon
  */
-void Irohad::init() {
-  // Recover WSV from the existing ledger to be sure it is consistent
-  initWsvRestorer();
-  restoreWsv();
-  if (opt_alternative_peers_) {
-    if (not resetPeers(*opt_alternative_peers_))
-      return;
-  }
-
-  initCryptoProvider();
-  initBatchParser();
-  initValidators();
-  initNetworkClient();
-  initFactories();
-  initPersistentCache();
-  initOrderingGate();
-  initSimulator();
-  initConsensusCache();
-  initBlockLoader();
-  initConsensusGate();
-  initSynchronizer();
-  initPeerCommunicationService();
-  initStatusBus();
-  initMstProcessor();
-  initPendingTxsStorage();
+Irohad::RunResult Irohad::init() {
+  auto peers_updater = [&]() -> RunResult {
```

Comment:
Why a separate lambda is required if the body can be put directly in lambda after `restoreWsv`?